python/ray/dag/compiled_dag_node.py

```diff
@@ -808,22 +859,21 @@ def _get_or_compile(
                 # `readers` is the nodes that are ordered after the current one (`task`)
                 # in the DAG.
                 readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
-
+                reader_to_node: List[Tuple["ray.actor.ActorHandle", str]] = []
                 dag_nodes = [reader.dag_node for reader in readers]
-                read_by_driver = False
+                read_by_output_node = False
                 for dag_node in dag_nodes:
                     if isinstance(dag_node, MultiOutputNode):
-                        read_by_driver = True
+                        read_by_output_node = True
                         break
-                if read_by_driver:
+                if read_by_output_node:
                     if len(readers) != 1:
                         raise ValueError(
                             "DAG outputs currently can only be read by the driver--not "
                             "the driver and actors."
                         )
-                    # This node is a multi-output node, which means that it will only be
-                    # read by the driver, not an actor. Thus, we handle this case by
-                    # setting `reader_handles` to `[self._driver_actor]`.
+                    # This node is a multi-output node, which means it will only be
+                    # read by the driver or the actor that is also the InputNode.
 
                     # TODO(jhumphri): Handle case where there is an actor, other than
```

Comment:
is this comment still relevant?

Reply:
Yes, it is still relevant. The comment applies to the case where a DAG node is read by both the driver process and another actor. For example, `a.inc` is read by actor b and the driver process at the same time. This example doesn't work currently.

```
driver -> a.inc -> b.inc -> driver 
                |        |
                ----------
```